File: Bogosort/BogoSort-Go/BogoSort.go
package main

import (
	"fmt"
	"math/rand"
)

// custom type array
type MyArray []int64

func (arr MyArray) BogoSort() {
	for !arr.isSorted() {
		for i := 0; i < len(arr); i++ {
			randomIndex := rand.Intn(len(arr))
			arr[i], arr[randomIndex] = arr[randomIndex], arr[i]
		}
	}
}

func (arr MyArray) isSorted() bool {
	for i := 0; i < len(arr)-1; i++ {
		if arr[i] > arr[i+1] {
			return false
		}
	}
	return true
}

func main() {
	arr := MyArray{3, 1, 2}
	fmt.Printf("Before sort: %v\n", arr) //Before sort: [3 1 2]
	arr.BogoSort()
	fmt.Printf("After sort: %v\n", arr) //After sort: [1 2 3]
}

File: BogoSort-Java/BogoSort.java
import java.util.*;

public class BogoSort {
   private static final Random RAN_OUTPUT = new Random();
   
    public static void bogoSort(int[] arr) {
      while (isSorted(arr) == false) {
         for (int i = 0; i < arr.length; i++) {
            int randomIndex = RAN_OUTPUT.nextInt(arr.length);
            int holder = arr[i];
            arr[i] = arr[randomIndex];
            arr[randomIndex] = holder;
         }
      }
    }
    
    private static boolean isSorted(int[] arr) {
      for (int i = 0; i < arr.length - 1; i++) {
         if (arr[i] > arr[i+1]) {
            return false;
         } 
      }
      return true;
    }
    
    public static void main(String[] args) {
      int[] arr = {3, 1, 2};
      System.out.println("Before Sort: " + Arrays.toString(arr));
      bogoSort(arr);
      System.out.println("After Sort: " + Arrays.toString(arr));
    }
    
}